barricade/src/main/java/com/mutualmobile/barricade/utils/BarricadeShakeListener.java

```diff
@@ -45,17 +51,62 @@ public BarricadeShakeListener(Context context) {
       }
       if (shakeCount >= 2) {
         shakeCount = 0;
-        Intent intent = new Intent(mContext, BarricadeActivity.class);
+        Intent intent = new Intent(application, BarricadeActivity.class);
         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-        mContext.startActivity(intent);
+        application.startActivity(intent);
       }
       lastX = x;
       lastY = y;
       lastZ = z;
     }
   }
 
+  private void enableShakeListener() {
+    sensorManager.registerListener(this, sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER),
+        SensorManager.SENSOR_DELAY_GAME);
+  }
+
+  private void disableShakeListener() {
+    sensorManager.unregisterListener(this);
+  }
+
   @Override public void onAccuracyChanged(Sensor sensor, int i) {
 
   }
+
+  @Override public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+
+  }
+
+  @Override public void onActivityStarted(Activity activity) {
+    if (activityCount == 0) {
+      enableShakeListener();
+    }
+
+    activityCount++;
+  }
```

Comment:
Just for better readability, it's better if we put  `activityCount++` before condition check of if (activityCount == 1)... Since we followed the same structure for inActivityStopped